src/WebJobs.Script/Rpc/LanguageWorkerChannel.cs

```diff
@@ -178,36 +178,60 @@ public void SetupFunctionInvocationBuffers(IEnumerable<FunctionMetadata> functio
             }
         }
 
-        public void SendFunctionLoadRequests()
+        public async Task SendFunctionLoadRequests()
         {
-            if (_functions != null)
+            var initialized = await _workerInitTask.Task;
+
+            if (_functions != null && initialized)
             {
                 foreach (FunctionMetadata metadata in _functions)
                 {
                     SendFunctionLoadRequest(metadata);
                 }
             }
+            else
+            {
+                if (_functions == null)
+                {
+                    _workerChannelLogger.LogDebug("Not sending function load requests. No functions found.");
+                }
+                else if (!initialized)
+                {
+                    _workerChannelLogger.LogDebug("Not sending function load requests. Worker was able to initialize");
+                }
+            }
         }
 
-        public Task SendFunctionEnvironmentReloadRequest()
+        public async Task<bool> SendFunctionEnvironmentReloadRequest()
         {
-            _workerChannelLogger.LogDebug("Sending FunctionEnvironmentReloadRequest");
-            _eventSubscriptions
-                .Add(_inboundWorkerEvents.Where(msg => msg.MessageType == MsgType.FunctionEnvironmentReloadResponse)
-                .Timeout(workerInitTimeout)
-                .Take(1)
-                .Subscribe((msg) => FunctionEnvironmentReloadResponse(msg.Message.FunctionEnvironmentReloadResponse)));
+            var initialized = await _workerInitTask.Task;
```

Comment:
Prevents function environment reload (for specialization) from happening before worker is ready

Reply:
_workerInitTask happens during start up and env reload happens during specialization. Need to keep them separate